pkg/network/usm/nodejs.go

```diff
@@ -153,6 +157,7 @@ func (m *nodeJSMonitor) Stop() {
 		return
 	}
 
+	m.processMonitor.Stop()
```

Comment:
why not encapsulate that inside the upobe attacher? call the stop inside attacher.Stop(), that way you don't need to store the pm reference in the ebpf program

Reply:
For example, if we use a process monitor from the event data stream, the attacher cannot be the one calling `Stop()` anymore as it's not needed for the consumer-facing side (there's a `Stop()` function but that one is called by the event monitor itself). 

Even if that wasn't the case, the attacher it's not managing the process-monitor lifecycle anymore (it's now receiving an initialized/started monitor, not creating it) so I feel it shouldn't be the one calling `Stop()`, either we manage all the lifecycle or none of it.

Reply:
ok in that case add a comment to the NewUprobeAttacher that it is the responsibility of the caller to clean up any resources associated with the processMonitor after usage

Reply:
Added, also improved the comments on the constructor.